firefly-cardanoconnect/src/streams/mux.rs

```diff
@@ -533,3 +534,119 @@ impl StreamSubscription {
         self.batch_receiver.recv().await
     }
 }
+
+#[cfg(test)]
```

Comment:
ah you are written the tests in the same file!

Reply:
Yep, it's the most common way to write unit tests in rust.